### PR TITLE
Add modal transitions

### PR DESCRIPTION
**Pull request by nitriques**
This is a reference for future modal to modal transitions

It adds the new modal transition and also adds the modal-to-modal
transition.

### CODE REVIEW

**Comment by nitriques** [comment]
@DeuxHuitHuit/review ping

File: src/transitions/modal-transition.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+	
+	var defaultLeaveTransition = function (data, callback) {
+		loadCompleted = true;
+		dataIn = data;
+		if (beginCompleted) {
+			completeLeaveAnim(data, callback);
+		}
+	};
+	
+	/*App.transitions.exports({
```
Comment (fhamon) [inline src/transitions/modal-transition.js]: We should maybe clean this a bit one example is enough 😁

**Review by fhamon — CHANGES_REQUESTED** [review]

File: src/transitions/modal-transition.js (hunk 1 above)
Comment (nitriques) [reply src/transitions/modal-transition.js]: @dominic-blain  ?

File: src/transitions/modal-transition.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+			App.modules.notify('modal.prepare', {
+				item: modal
+			});
+			domEnteringPage.css({
+				opacity: 1,
+				display: 'block',
+				position: 'relative'
+			}).height();
+			enteringPage.enter(data.enterNext);
+			App.modules.notify('modal.show', {
```
Comment (dominic-blain) [inline src/transitions/modal-transition.js]: Is there a module that listens to modal.show/prepare/etc. ?

File: src/transitions/modal-transition.js [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+(function ($, undefined) {
+	
+	'use strict';
+	
+	var win = $(window);
+	var body = $('body');
+	var sitePages = $('#site-pages');
+	var bgTransition = $('#bg-transition');
+	var bgTransitionModal = $('#bg-transition-modal');
+	var DEFAULT_DELAY = 350;
```
Comment (PascalPiche) [inline src/transitions/modal-transition.js]: Unused variable

File: src/transitions/modal-to-modal-transition.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+ */
+(function ($, undefined) {
+	
+	'use strict';
+	
+	var win = $(window);
+	var body = $('body');
+	var sitePages = $('#site-pages');
+	var bgTransition = $('#bg-transition');
+	var bgTransitionModal = $('#bg-transition-modal');
```
Comment (PascalPiche) [inline src/transitions/modal-to-modal-transition.js]: unused variable

File: src/transitions/modal-transition.js [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+ */
+(function ($, undefined) {
+	
+	'use strict';
+	
+	var win = $(window);
+	var body = $('body');
+	var sitePages = $('#site-pages');
+	var bgTransition = $('#bg-transition');
+	var bgTransitionModal = $('#bg-transition-modal');
```
Comment (PascalPiche) [inline src/transitions/modal-transition.js]: Unused variable

File: src/transitions/modal-to-modal-transition.js [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+ *
+ */
+(function ($, undefined) {
+	
+	'use strict';
+	
+	var win = $(window);
+	var body = $('body');
+	var sitePages = $('#site-pages');
+	var bgTransition = $('#bg-transition');
```
Comment (PascalPiche) [inline src/transitions/modal-to-modal-transition.js]: Unused variable

File: src/transitions/modal-transition.js [hunk 7]
```diff
@@ -0,0 +1,249 @@
...
+ *
+ */
+(function ($, undefined) {
+	
+	'use strict';
+	
+	var win = $(window);
+	var body = $('body');
+	var sitePages = $('#site-pages');
+	var bgTransition = $('#bg-transition');
```
Comment (PascalPiche) [inline src/transitions/modal-transition.js]: Unused variable

File: src/transitions/modal-to-modal-transition.js [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+ * Modal page to modal page transition
+ *
+ */
+(function ($, undefined) {
+	
+	'use strict';
+	
+	var win = $(window);
+	var body = $('body');
+	var sitePages = $('#site-pages');
```
Comment (PascalPiche) [inline src/transitions/modal-to-modal-transition.js]: Unsued variable

File: src/transitions/modal-transition.js [hunk 9]
```diff
@@ -0,0 +1,249 @@
...
+		
+		//Check if the enteringPage is the background page
+		if (!isModalReturn) {
+			//Remove the other pages
+			var oldBgPage = sitePages.find('.is-bg-modal-page');
+			if (oldBgPage.length) {
+				body.removeClass(oldBgPage.attr('id'));
+				oldBgPage.remove();
+			}
+
```
Comment (PascalPiche) [inline src/transitions/modal-transition.js]: We are missing a remove domEnteringPage.removeClass('is-bg-modal-page');

**Review by PascalPiche — CHANGES_REQUESTED** [review]

File: src/transitions/modal-transition.js (hunk 1 above)
Comment (fhamon) [reply src/transitions/modal-transition.js]: I say we let one example, right now their's 2 of them.

Comment (dominic-blain) [reply src/transitions/modal-transition.js]: Yes, my point exactly

File: src/transitions/modal-transition.js (hunk 2 above)
Comment (PascalPiche) [reply src/transitions/modal-transition.js]: Need to validate. all the transition in this file seem depend on an other module.

Comment (dominic-blain) [reply src/transitions/modal-transition.js]: Checked and is in auto-modal module.

File: src/transitions/modal-transition.js [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+	
+	'use strict';
+	
+	var win = $(window);
+	var body = $('body');
+	var sitePages = $('#site-pages');
+	var bgTransition = $('#bg-transition');
+	var bgTransitionModal = $('#bg-transition-modal');
+	var DEFAULT_DELAY = 350;
+	var MODAL_SELECTOR = '.js-modal-page';
```
[the remark this answers is not in the document]
Comment (dominic-blain) [reply src/transitions/modal-transition.js]: Done

**Comment by dominic-blain** [comment]
Added in starter kit as-is. Will update when PR is merged

**Pull request closed without merging**